File: NotificationSystem/NotificationSystem.Web/ctrTroubleTicketReq.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.Data;

namespace NotificationSystem.NotificationSystem.Web
{
    public partial class ctrTroubleTicketReq : System.Web.UI.UserControl
    {

        private int m_TroubleTicketNo = 0;

        public int TroubleTicketNo
        {
            get
            {
                return m_TroubleTicketNo;
            }
            set
            {
                m_TroubleTicketNo = value;
            }
        }
		protected void Page_Load(object sender, EventArgs e)
		{


            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
            TroubleTicketReqDataTable tblTTR = new TroubleTicketReqDataTable();

            try
            {
                if (Request.Form["ctl00$MainContent$ctrTroubleTicketReq_Search$btnSearch"] == "Search")
                    this.ctrHiddebField.Value = Convert.ToString(m_TroubleTicketNo);


                if ((Page.IsPostBack) & this.ctrHiddebField.Value.Length > 0)
					tblTTR = (TroubleTicketReqDataTable)theNotificationSystem.GetTroubleTicket(m_TroubleTicketNo);
                else
                    tblTTR = (TroubleTicketReqDataTable)theNotificationSystem.GetTroubleTicket();

                this.lblSearchResult.Text = tblTTR.Rows.Count + " Result(s)";
                this.grdTroubleTicketReq.DataSource = tblTTR.DefaultView;
                this.grdTroubleTicketReq.DataBind();
            }
            catch (Exception)
            {
                throw;
            }
        }
		
        private string ConvertSortDirection(System.Web.UI.WebControls.GridViewSortEventArgs e)
        {
            ViewState.Add("columnname", e.SortExpression);

            if ((ViewState["direction"] == null))
                ViewState.Add("direction", "asc");
            else
                ViewState["direction"] = Interaction.IIf(ViewState["direction"].ToString().ToLower() == "desc", "asc", "desc");

            return ViewState["direction"].ToString();
        }

		protected void grdTroubleTicketReq_PageIndexChanging(object sender, GridViewPageEventArgs e)
		{
            try
            {
                if (!(Information.IsNothing(ViewState["columnname"]) | Information.IsNothing(ViewState["direction"])))
                {
                    DataView m_DataView = (DataView)this.grdTroubleTicketReq.DataSource;

                    if (m_DataView == null)
                    {
                        m_DataView.Sort = ViewState["columnname"].ToString() + " " + ViewState["direction"].ToString();
                        this.grdTroubleTicketReq.DataSource = m_DataView;
                    }
                }

                this.grdTroubleTicketReq.PageIndex = e.NewPageIndex;
                this.grdTroubleTicketReq.DataBind();
            }
            catch (Exception)
            {
                throw;
            }
        }

		protected void grdTroubleTicketReq_Sorting(object sender, GridViewSortEventArgs e)
		{
            try
            {
                DataView m_Dataview = (DataView)grdTroubleTicketReq.DataSource;

                if (m_Dataview == null)
                {
                    m_Dataview.Sort = e.SortExpression + " " + ConvertSortDirection(e);
                    this.grdTroubleTicketReq.DataSource = m_Dataview;
                    this.grdTroubleTicketReq.DataBind();
                }
            }
            catch (Exception)
            {
                throw;
            }
        }
	}
}

File: NotificationSystem/NotificationSystem.Web/ctrTroubleTicketReq.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports Microsoft.VisualBasic
Imports System.Data

Namespace NotificationSystem.NotificationSystem.Web
    Public Partial Class ctrTroubleTicketReq
        Inherits UserControl

        Private m_TroubleTicketNo As Integer = 0

        Public Property TroubleTicketNo As Integer
            Get
                Return m_TroubleTicketNo
            End Get
            Set(value As Integer)
                m_TroubleTicketNo = value
            End Set
        End Property
        Protected Sub Page_Load(sender As Object, e As EventArgs)


            Dim theNotificationSystem As NotificationSystem.NotificationSystem.Data.Classes.clsNotificationSystem = New NotificationSystem.NotificationSystem.Data.Classes.clsNotificationSystem()
            Dim tblTTR As NotificationSystem.NotificationSystem.Data.NotificationSystem.TroubleTicketReqDataTable = New NotificationSystem.NotificationSystem.Data.NotificationSystem.TroubleTicketReqDataTable()

            Try
                If Equals(Request.Form("ctl00$MainContent$ctrTroubleTicketReq_Search$btnSearch"), "Search") Then ctrHiddebField.Value = Convert.ToString(m_TroubleTicketNo)


                If MyBase.Page.IsPostBack And ctrHiddebField.Value.Length > 0 Then
                    tblTTR = CType(theNotificationSystem.GetTroubleTicket(m_TroubleTicketNo), NotificationSystem.NotificationSystem.Data.NotificationSystem.TroubleTicketReqDataTable)
                Else
                    tblTTR = CType(theNotificationSystem.GetTT(), NotificationSystem.NotificationSystem.Data.NotificationSystem.TroubleTicketReqDataTable)
                End If

                lblSearchResult.Text = tblTTR.Rows.Count.ToString() & " Result(s)"
                grdTroubleTicketReq.DataSource = tblTTR.DefaultView
                grdTroubleTicketReq.DataBind()
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub

        Private Function ConvertSortDirection(e As GridViewSortEventArgs) As String
            MyBase.ViewState.Add("columnname", e.SortExpression)

            If MyBase.ViewState("direction") Is Nothing Then
                MyBase.ViewState.Add("direction", "asc")
            Else
                MyBase.ViewState("direction") = IIf(Equals(MyBase.ViewState("direction").ToString().ToLower(), "desc"), "asc", "desc")
            End If

            Return MyBase.ViewState("direction").ToString()
        End Function

        Protected Sub grdTroubleTicketReq_PageIndexChanging(sender As Object, e As GridViewPageEventArgs)
            Try
                If Not (IsNothing(MyBase.ViewState("columnname")) Or IsNothing(MyBase.ViewState("direction"))) Then
                    Dim m_DataView = CType(grdTroubleTicketReq.DataSource, DataView)

                    If m_DataView Is Nothing Then
                        m_DataView.Sort = MyBase.ViewState("columnname").ToString() & " " & MyBase.ViewState("direction").ToString()
                        grdTroubleTicketReq.DataSource = m_DataView
                    End If
                End If

                grdTroubleTicketReq.PageIndex = e.NewPageIndex
                grdTroubleTicketReq.DataBind()
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub

        Protected Sub grdTroubleTicketReq_Sorting(sender As Object, e As GridViewSortEventArgs)
            Try
                Dim m_Dataview = CType(grdTroubleTicketReq.DataSource, DataView)

                If m_Dataview Is Nothing Then
                    m_Dataview.Sort = e.SortExpression & " " & ConvertSortDirection(e)
                    grdTroubleTicketReq.DataSource = m_Dataview
                    grdTroubleTicketReq.DataBind()
                End If
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub
    End Class
End Namespace
